playkit/src/main/java/com/kaltura/playkit/player/PlayerController.java

```diff
@@ -93,8 +93,13 @@ public void onEvent(PlayerEvent.Type eventType) {
                 // TODO: use specific event class
                 switch (eventType) {
                     case DURATION_CHANGE:
-                        event = new PlayerEvent.DurationChanged(getDuration());
-                        if (getDuration() != Consts.TIME_UNSET && isNewEntry) {
+                        long newDuration = getDuration();
+                        if (newDuration == Consts.TIME_UNSET) {
```

Comment:
Masking this event requires careful testing, someone could rely on it. Also must be compared to iOS.